conans/test/integration/command_v2/test_graph_find_binaries.py

```diff
@@ -124,6 +170,44 @@ def test_different_setting(self, client):
         assert pkg1["diff"]["dependencies"] == {}
         assert pkg1["diff"]["explanation"] == "This binary was built with different settings."
 
+    def test_different_settings_target(self):
+        c = TestClient()
+        conanfile = textwrap.dedent("""\
+            from conan import ConanFile
+            class Pkg(ConanFile):
+                name = "tool"
+                version = "1.0"
+                def package_id(self):
+                    self.info.settings_target = self.settings_target.copy()
+                    self.info.settings_target.constrained(["os"])
```

Comment:
TIL about contstrained!

Reply:
We need to discuss and document better how to manage settings_target in ``package_id``, docs are wrong (modifying wihout a copy)